src/repository.ts

```diff
@@ -234,27 +234,46 @@ export class Repository {
     this.disposables.push(this.unversioned);
     this.disposables.push(this.conflicts);
 
-    const updateFreqNew = configuration.get<number>(
-      "remoteChanges.checkFrequency",
-      300
+    // setInterval id for remoteChanges.checkFrequency
+    let interval: NodeJS.Timer;
+
+    this.disposables.push(
+      toDisposable(() => {
+        clearInterval(interval);
+      })
     );
-    if (updateFreqNew) {
-      const interval = setInterval(() => {
-        this.updateRemoteChangedFiles();
-      }, 1000 * updateFreqNew);
 
-      this.disposables.push(
-        toDisposable(() => {
-          clearInterval(interval);
-        })
+    // Update the interval of frequency of remote changes
+    const checkUpdateFreq = () => {
+      const updateFreqNew = configuration.get<number>(
```

Comment:
This cant be a const if it overwritten. Aslo it will only have scopt in the function so outside will have no visibility

Reply:
Your suggestion?

Reply:
You just need to declare `updateFreqNew` outside that function

Reply:
Changed to attribute of Repository